src/app/modules/consultations/consultation-profile/consultation-profile.component.ts

```diff
@@ -67,6 +70,8 @@ export class ConsultationProfileComponent implements OnInit, OnDestroy {
         this.responseList = data.sharedResponses.edges;
     }, err => {
       this.errorService.showErrorModal(err);
+      this.cookieService.set('loginCallbackUrl', this.router.url);
```

Comment:
@Harshwardhan18 This will not work. What this will do is set the callbackUrl and redirect the user for all API errors. You need to specifically check for a 401 error and only then set the callbackUrl and redirect.

Reply:
@Harshwardhan18 You can pass this `err` object to Error constants file(already there) as 
`const error = new Error(err);`
 and you'll get an object with error message. Check for the message `Invalid Access Token`, then do the thing

Reply:
@Harshwardhan18 In that case don't show the error message to the user `this.errorService.showErrorModal(err)` show it only if it is not that message.

Reply:
@Manzurkds bro here they've used a generalised error modal. Even if I use if statement it won't work as we want